Fork_Maintenance.md

```diff
@@ -0,0 +1,263 @@
+# OpenHands Fork Maintenance Guide
+
+This guide outlines a strategy for maintaining a fork of the main OpenHands repository while incorporating custom patches and ensuring stability and security.
+
+## Key Goals
+
+1.  **Consistent Container Images:** Use a specific, matched pair of `runtime` and `agent` images across all environments.
+2.  **Security & Intentional Updates:** Avoid automatic updates from upstream; intentionally review and integrate changes.
+3.  **Local Patch Management:** Easily add, manage, and remove custom features or external PRs.
+
+## 1. Initial Setup
+
+First, ensure your fork is cloned locally and the original OpenHands repository is added as a remote named `upstream`.
+
+```bash
+# Clone your fork (replace with your fork URL)
+# git clone <your-fork-url>
+# cd OpenHands
+
+# Add the main OpenHands repository as the 'upstream' remote
+git remote add upstream https://github.com/All-Hands-AI/OpenHands.git
+
+# Fetch upstream branches and tags
+git fetch upstream
+```
+
+## 2. Container Image Management
+
+To ensure consistency, build, tag, and distribute your own `runtime` and `agent` images based on your primary integration branch.
+
+**a. Building Images:**
+
+*   **Source Branch:** Images should be built by CI/CD processes triggered by commits to the `release/stable-with-patches` branch.
+*   **Build Process:** Use the standard OpenHands build process (`docker build`, `make`, etc.).
+*   **Base Image Consideration:** While the image is built *from* the code in `release/stable-with-patches`, consider how the underlying base OS/dependency image specified in the Dockerfile is kept up-to-date. This might involve periodically updating the base image tag in the Dockerfile itself as part of the fork maintenance.
```

Comment:
I think this looks like having either an implicit or explicit "base commit" for the entire fork.

I.e. our fork may be `0.33` + some patches. When `0.34` is released, we can update it to look exactly like `0.34` + some patches. If we do that right, we can programmatically determine the commit we're based on (since that commit will be a part of git history) and use that same tag for the base in our docker images.

I feel like this should already be a solved problem, but I am not familiar with anything that keeps this enforced and simple. Anyone ever done this before? My guess is that you just use a build arg in the dockerfile and pass in the base image for the commit we're built on.